pom.xml

```
@@ -261,72 +263,15 @@
-          <version>2.22.2</version>
-          <configuration>
-            <trimStackTrace>false</trimStackTrace>
-            <argLine>${java.test.argLine}</argLine>
-          </configuration>
-          <executions>
-            <execution>
-              <goals>
-                <goal>integration-test</goal>
-                <goal>verify</goal>
-              </goals>
-            </execution>
-          </executions>
-        </plugin>
-        <plugin>
-          <groupId>org.apache.maven.plugins</groupId>
-          <artifactId>maven-surefire-plugin</artifactId>
-          <version>2.22.2</version>
-          <configuration>
-            <trimStackTrace>false</trimStackTrace>
-            <argLine>${java.test.argLine}</argLine>
-          </configuration>
-        </plugin>
-        <plugin>
-          <groupId>org.sonatype.plugins</groupId>
-          <artifactId>nexus-staging-maven-plugin</artifactId>
-          <version>1.6.1</version>
-          <configuration>
-            <nexusUrl>http://nexus.terracotta.eur.ad.sag</nexusUrl>
-            <serverId>terracotta-nexus-staging</serverId>
-            <skipNexusStagingDeployMojo>${skipDeploy}</skipNexusStagingDeployMojo>
-          </configuration>
-        </plugin>
-        <plugin>
-          <groupId>org.apache.maven.plugins</groupId>
-          <artifactId>maven-toolchains-plugin</artifactId>
-          <version>3.0.0</version>
-          <configuration>
-            <toolchains>
-              <jdk>
-                <version>${java.build.version}</version>
-              </jdk>
-            </toolchains>
-          </configuration>
-          <executions>
-            <execution>
-              <goals>
-                <goal>toolchain</goal>
-              </goals>
-            </execution>
-          </executions>
-        </plugin>
-        <plugin>
+          <inherited>false</inherited>
           <groupId>com.mycila</groupId>
           <artifactId>license-maven-plugin</artifactId>
```

<artifactId>license-maven-plugin</artifactId> could be moved to the plugin management section of the profile

---

@mathieucarbou I see two parts to this plugin, one in pluginManagement and the other in plugins.  The part in plugins merely specifies an execution, but it also has a suspicious <inherited>false ... Documentation suggests that "check" is automatically bound to `verify`, so I don't know why any of this is in the pom in the first place.  Do you know?

---

yesI did this plugin ;-) `<inherited>false<inherited>` is required for the license plugin in case you have a multi-module project and you want to execute it only once form the paent pom across all the project files. This is also required because of that: `<header>header.txt</header>` If the plugin executes in some sub-modules, then it won't find the header.
The license plugin has some pre-defined headers already, APACHE being suppported, but sadly we are overriting the apache license to add: `Copyright Terracotta, Inc.`. I don't even know if this is legal to alter the apache license like that... I think it breaks its compatibility becasue of the copyright.
The license plugin is also able to take teh license from any URI or even a classpath, when license is put in a jar as a plugin dependency.
So depending how it is configured and which type of project, inherited=false is required or not.
For our OSS projects, I think we should in theory go with the normal apache supported license without providing our own. I don't know where this one comes from but what is sure is that Copyrighting a source file as it is one in the current license is not compatible with the terms of the apache license.

---

@akomakom to add, here are the license of plugins plus template that comes with the license plugin: https://github.com/mycila/license-maven-plugin/blob/master/license-maven-plugin/src/main/resources/com/mycila/maven/plugin/license/templates/. The template already supports some placeholders.
Can be accessed like that:
`<header>com/mycila/maven/plugin/license/templates/APACHE-2.txt</header>`

---

I have no idea whether we can switch from a tc license to apache, but perhaps we can move this plugin definition to terracotta-parent (using tc-core definition as it appears to have a superset of exclusions)?

---

yes this plugin is really configured per project.
But in the parent pom, you can enforce its execution though, and set an id to the execution section: in case a sub project wants to disable the plugin execution, he will be able to thanks to the id.

---

So I'm looking at platform and core... and core doesn't appear to declare any execution for the license plugin at this time.  So either it's not necessary, or it doesn't work in core (I'm guessing - the latter)

---

It is because tc-platform project has been created more recently - we added license plugin with Anthony. core project was already there. But perhaps Myron/Sairam would be interested in having that because they sometimes commit files without headers I saw ;-)

---

What I meant was that core [does mention the license plugin](https://github.com/Terracotta-OSS/terracotta-core/blob/master/pom.xml#L87) but does not define any executions.  I just removed a license header from source and it still compiled :)

---

... which is one of the reasons I want to move common elements to parent, so that things are not easily overlooked and do not fall out of sync.  (from an infra perspective).

---

@mathieucarbou @myronkscott should license checking happen at:
* compile (fail fast)
* verify  (check any possible generated files)

---

I designed the plugin for an execution during verify phase. And generated files (in target folder) should be excluded from the check: there are some default exclusions for everything inside target/**. Except if explicitly changed.

---

Now I'm puzzled.  It seems to me that license check at compile would be the same as license check at verify with exclusions... except that there won't need to be any exclusions... unless I'm misunderstanding.

---

Here is my plan: https://github.com/Terracotta-OSS/terracotta-parent/pull/13  Exclusions seem to work for core and platform, and I'll be making a PR for core because it never had license plugin enabled until now, so half the headers were wrong or missing.  Currently license execution is bound to `compile`, but I'm not married to that.

---

put some comments over there ;-)